src/main/java/de/tum/in/www1/artemis/service/BuildLogEntryService.java

```diff
@@ -130,4 +131,45 @@ public boolean checkIfBuildLogIsNotADuplicate(ProgrammingLanguage programmingLan
         }
         return true;
     }
+
+    /**
+     * Filters out unnecessary build logs that a student should not see.
+     *
+     * @param buildLogEntries the build logs
+     * @param programmingLanguage the programming language
+     * @return filtered build logs
+     */
+    public List<BuildLogEntry> removeUnnecessaryLogsForProgrammingLanguage(List<BuildLogEntry> buildLogEntries, ProgrammingLanguage programmingLanguage) {
+        List<BuildLogEntry> filteredLogs = new ArrayList<>();
+        for (BuildLogEntry buildLog : buildLogEntries) {
+            boolean compilationErrorFound = false;
+            String logString = buildLog.getLog();
+
+            if (logString.contains("COMPILATION ERROR")) {
+                compilationErrorFound = true;
+            }
+
+            if (compilationErrorFound && logString.contains("BUILD FAILURE")) {
+                // hide duplicated information that is displayed in the section COMPILATION ERROR and in the section BUILD FAILURE and stop here
```

Comment:
Same as above. Maybe create a method `isBuildFailure` that actually compares the strings to provide the possibility to change the code later :)